Error while checking build/O2/fullCI for b6c728dedf26a248210b6ce2e27369d72d4d7587 at 2023-03-10 19:21:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:620:40: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:622:16: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:637:42: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:639:18: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:672:58: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:674:58: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:676:58: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:678:58: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:691:24: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:693:9: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/10932-slc8_x86-64/0/Detectors/EMCAL/base/src/ClusterFactory.cxx:695:9: error: statement should be inside braces [readability-braces-around-statements]
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10932/b6c728dedf26a248210b6ce2e27369d72d4d7587/build_O2_fullCI/pretty.html).

--- a/Detectors/EMCAL/base/include/EMCALBase/ClusterFactory.h
+++ b/Detectors/EMCAL/base/include/EMCALBase/ClusterFactory.h
@@ -271,7 +271,7 @@ class ClusterFactory
 
   /// \brief Look to cell neighbourhood and reject if it seems exotic
   /// \param towerId: tower ID of cell with largest energy fraction in cluster
-  /// \param ecell: energy of the cell with largest energy fraction in cluster 
+  /// \param ecell: energy of the cell with largest energy fraction in cluster
   /// \param exoticTime time of the cell with largest energy fraction in cluster
   /// \return bool true if cell is found exotic
   bool isExoticCell(short towerId, float ecell, float const exoticTime) const;

--- a/Detectors/EMCAL/base/include/EMCALBase/ClusterFactory.h
+++ b/Detectors/EMCAL/base/include/EMCALBase/ClusterFactory.h
@@ -266,7 +266,27 @@ class ClusterFactory
   /// \return the index of the cells with max enegry
   /// \return the maximum energy
   /// \return the total energy of the cluster
-  std::tuple<int, float, float> getMaximalEnergyIndex(gsl::span<const int> inputsIndices) const;
+  /// \return if cluster is shared between super models
+  std::tuple<int, float, float, bool> getMaximalEnergyIndex(gsl::span<const int> inputsIndices) const;
+
+  /// \brief Look to cell neighbourhood and reject if it seems exotic
+  /// \param towerId: tower ID of cell with largest energy fraction in cluster
+  /// \param ecell: energy of the cell with largest energy fraction in cluster 
+  /// \param exoticTime time of the cell with largest energy fraction in cluster
+  /// \return bool true if cell is found exotic
+  bool isExoticCell(short towerId, float ecell, float const exoticTime) const;
+
+  /// \brief Calculate the energy in the cross around the energy of a given cell.
+  /// \param absID: controlled cell absolute ID number
+  /// \param energy: cluster or cell max energy, used for weight calculation
+  /// \param exoticTime time of the cell with largest energy fraction in cluster
+  /// \return the energy in the cross around the energy of a given cell
+  float getECross(short absID, float energy, float const exoticTime) const;
+
+  /// \param eCell: cluster cell energy
+  /// \param eCluster: cluster or cell max energy
+  /// \return weight of cell for shower shape calculation
+  float GetCellWeight(float eCell, float eCluster) const;
 
   ///
   /// Calculates the multiplicity of digits/cells with energy larger than level*energy
@@ -292,6 +312,21 @@ class ClusterFactory
 
   bool getCoreRadius() const { return mCoreRadius; }
   void setCoreRadius(float radius) { mCoreRadius = radius; }
+
+  float getExoticCellFraction() const { return mExoticCellFraction; }
+  void setExoticCellFraction(float exoticCellFraction) { mExoticCellFraction = exoticCellFraction; }
+
+  float getExoticCellDiffTime() const { return mExoticCellDiffTime; }
+  void setExoticCellDiffTime(float exoticCellDiffTime) { mExoticCellDiffTime = exoticCellDiffTime; }
+
+  float getExoticCellMinAmplitude() const { return mExoticCellMinAmplitude; }
+  void setExoticCellMinAmplitude(float exoticCellMinAmplitude) { mExoticCellMinAmplitude = exoticCellMinAmplitude; }
+
+  float getExoticCellInCrossMinAmplitude() const { return mExoticCellInCrossMinAmplitude; }
+  void setExoticCellInCrossMinAmplitude(float exoticCellInCrossMinAmplitude) { mExoticCellInCrossMinAmplitude = exoticCellInCrossMinAmplitude; }
+
+  bool getUseWeightExotic() const { return mUseWeightExotic; }
+  void setUseWeightExotic(float useWeightExotic) { mUseWeightExotic = useWeightExotic; }
 
   void setClustersContainer(gsl::span<const o2::emcal::Cluster> clusterContainer)
   {
@@ -358,9 +393,14 @@ class ClusterFactory
 
   bool mJustCluster = kFALSE; ///< Flag to evaluates local to "tracking" c.s. transformation (B.P.).
 
-  mutable int mSuperModuleNumber = 0; ///<  number identifying supermodule containing cluster, reference is cell with maximum energy.
-  float mDistToBadTower = -1;         ///<  Distance to nearest bad tower
-  bool mSharedCluster = false;        ///<  States if cluster is shared by 2 SuperModules in same phi rack (0,1), (2,3) ... (10,11).
+  mutable int mSuperModuleNumber = 0;         ///<  number identifying supermodule containing cluster, reference is cell with maximum energy.
+  float mDistToBadTower = -1;                 ///<  Distance to nearest bad tower
+  bool mSharedCluster = false;                ///<  States if cluster is shared by 2 SuperModules in same phi rack (0,1), (2,3) ... (10,11).
+  float mExoticCellFraction = 0.97;           ///<  Good cell if fraction < 1-ecross/ecell
+  float mExoticCellDiffTime = 1e6;            ///<  If time of candidate to exotic and close cell is too different (in ns), it must be noisy, set amp to 0
+  float mExoticCellMinAmplitude = 4.;         ///<  Check for exotic only if amplitud is larger than this value
+  float mExoticCellInCrossMinAmplitude = 0.1; ///<  Minimum energy of cells in cross, if lower not considered in cross
+  bool mUseWeightExotic = false;              ///<  States if weights should be used for exotic cell cut
 
   gsl::span<const o2::emcal::Cluster> mClustersContainer; ///< Container for all the clusters in the event
   gsl::span<const InputType> mInputsContainer;            ///< Container for all the cells/digits in the event

--- a/Detectors/EMCAL/base/src/ClusterFactory.cxx
+++ b/Detectors/EMCAL/base/src/ClusterFactory.cxx
@@ -73,13 +73,19 @@ o2::emcal::AnalysisCluster ClusterFactory<InputType>::buildCluster(int clusterIn
   // First calculate the index of input with maximum amplitude and get
   // the supermodule number where it sits.
 
-  auto [inputIndMax, inputEnergyMax, cellAmp] = getMaximalEnergyIndex(inputsIndices);
+  auto [inputIndMax, inputEnergyMax, cellAmp, shared] = getMaximalEnergyIndex(inputsIndices);
+
+  short towerId = mInputsContainer[inputIndMax].getTower();
+
+  float exoticTime = mInputsContainer[inputIndMax].getTimeStamp();
+
+  clusterAnalysis.setIsExotic(isExoticCell(towerId, inputEnergyMax, exoticTime));
 
   clusterAnalysis.setIndMaxInput(inputIndMax);
 
   clusterAnalysis.setE(cellAmp);
 
-  mSuperModuleNumber = mGeomPtr->GetSuperModuleNumber(mInputsContainer[inputIndMax].getTower());
+  mSuperModuleNumber = mGeomPtr->GetSuperModuleNumber(towerId);
 
   clusterAnalysis.setNCells(inputsIndices.size());
 
@@ -103,13 +109,13 @@ o2::emcal::AnalysisCluster ClusterFactory<InputType>::buildCluster(int clusterIn
   evalTime(inputsIndices, clusterAnalysis);
 
   // TODO to be added at a later stage
-  //evalPrimaries(inputsIndices, clusterAnalysis);
-  //evalParents(inputsIndices, clusterAnalysis);
+  // evalPrimaries(inputsIndices, clusterAnalysis);
+  // evalParents(inputsIndices, clusterAnalysis);
 
   // TODO to be added at a later stage
   // Called last because it sets the global position of the cluster?
   // Do not call it when recalculating clusters out of standard reconstruction
-  //if (!mJustCluster)
+  // if (!mJustCluster)
   //  evalLocal2TrackingCSTransform();
 
   return clusterAnalysis;
@@ -215,7 +221,7 @@ void ClusterFactory<InputType>::evalLocalPosition(gsl::span<const int> inputsInd
       continue;
     }
 
-    //Temporal patch, due to mapping problem, need to swap "y" in one of the 2 SM, although no effect in position calculation. GCB 05/2010
+    // Temporal patch, due to mapping problem, need to swap "y" in one of the 2 SM, although no effect in position calculation. GCB 05/2010
     if (mSharedCluster && mSuperModuleNumber != mGeomPtr->GetSuperModuleNumber(mInputsContainer[iInput].getTower())) {
       xyzi[1] *= -1;
     }
@@ -372,7 +378,7 @@ void ClusterFactory<InputType>::evalLocalPositionFit(double deff, double mLogWei
         clRmsXYZ[i] += (w * xyzi[i] * xyzi[i]);
       }
     }
-  } //loop
+  } // loop
 
   //  cout << " wtot " << wtot << endl;
 
@@ -407,7 +413,7 @@ void ClusterFactory<InputType>::evalLocalPositionFit(double deff, double mLogWei
     // May be put to global level or seperate method
     double ycorr = clXYZ[1] * (1. + phiSlope);
 
-    //printf(" y %f : ycorr %f : slope %f \n", clXYZ[1], ycorr, phiSlope);
+    // printf(" y %f : ycorr %f : slope %f \n", clXYZ[1], ycorr, phiSlope);
     clXYZ[1] = ycorr;
   }
 
@@ -533,7 +539,7 @@ void ClusterFactory<InputType>::evalElipsAxis(gsl::span<const int> inputsIndices
 
     lambda[1] = 0.5 * (dxx + dzz) - TMath::Sqrt(0.25 * (dxx - dzz) * (dxx - dzz) + dxz * dxz);
 
-    if (lambda[1] > 0) { //To avoid exception if numerical errors lead to negative lambda.
+    if (lambda[1] > 0) { // To avoid exception if numerical errors lead to negative lambda.
       lambda[1] = TMath::Sqrt(lambda[1]);
     } else {
       lambda[1] = 0.;
@@ -551,24 +557,158 @@ void ClusterFactory<InputType>::evalElipsAxis(gsl::span<const int> inputsIndices
 /// Finds the maximum energy in the cluster and computes the Summed amplitude of digits/cells
 //____________________________________________________________________________
 template <class InputType>
-std::tuple<int, float, float> ClusterFactory<InputType>::getMaximalEnergyIndex(gsl::span<const int> inputsIndices) const
+std::tuple<int, float, float, bool> ClusterFactory<InputType>::getMaximalEnergyIndex(gsl::span<const int> inputsIndices) const
 {
 
   float energy = 0.;
   int mid = 0;
   float cellAmp = 0;
+  int iSupMod0 = -1;
+  bool shared = false;
   for (auto iInput : inputsIndices) {
     if (iInput >= mInputsContainer.size()) {
       throw CellIndexRangeException(iInput, mInputsContainer.size());
     }
     cellAmp += mInputsContainer[iInput].getEnergy();
+    if (iSupMod0 == -1) {
+      iSupMod0 = mGeomPtr->GetSuperModuleNumber(mInputsContainer[iInput].getTower());
+    } else if (iSupMod0 != mGeomPtr->GetSuperModuleNumber(mInputsContainer[iInput].getTower())) {
+      shared = true;
+    }
     if (mInputsContainer[iInput].getEnergy() > energy) {
       energy = mInputsContainer[iInput].getEnergy();
       mid = iInput;
     }
   } // loop on cluster inputs
 
-  return std::make_tuple(mid, energy, cellAmp);
+  return std::make_tuple(mid, energy, cellAmp, shared);
+}
+
+///
+/// Look to cell neighbourhood and reject if it seems exotic
+//____________________________________________________________________________
+template <class InputType>
+bool ClusterFactory<InputType>::isExoticCell(short towerId, float ecell, float const exoticTime) const
+{
+  if (ecell < mExoticCellMinAmplitude) {
+    return false; // do not reject low energy cells
+  }
+
+  float eCross = getECross(towerId, ecell, exoticTime);
+
+  if (1 - eCross / ecell > mExoticCellFraction) {
+    LOG(debug) << "EXOTIC CELL id " << towerId << ", eCell " << ecell << ", eCross " << eCross << ", 1-eCross/eCell " << 1 - eCross / ecell;
+    return true;
+  }
+
+  return false;
+}
+
+///
+///  Calculate the energy in the cross around the energy of a given cell.
+//____________________________________________________________________________
+template <class InputType>
+float ClusterFactory<InputType>::getECross(short towerId, float energy, float const exoticTime) const
+{
+  auto [iSM, iMod, iIphi, iIeta] = mGeomPtr->GetCellIndex(towerId);
+  auto [iphi, ieta] = mGeomPtr->GetCellPhiEtaIndexInSModule(iSM, iMod, iIphi, iIeta);
+
+  // Get close cells index, energy and time, not in corners
+
+  short towerId1 = -1;
+  short towerId2 = -1;
+
+  if (iphi < o2::emcal::EMCAL_ROWS - 1)
+    towerId1 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi + 1, ieta);
+  if (iphi > 0)
+    towerId2 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi - 1, ieta);
+
+  // In case of cell in eta = 0 border, depending on SM shift the cross cell index
+
+  short towerId3 = -1;
+  short towerId4 = -1;
+
+  if (ieta == o2::emcal::EMCAL_COLS - 1 && !(iSM % 2)) {
+    towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM + 1, iphi, 0);
+    towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta - 1);
+  } else if (ieta == 0 && iSM % 2) {
+    towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta + 1);
+    towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM - 1, iphi, o2::emcal::EMCAL_COLS - 1);
+  } else {
+    if (ieta < o2::emcal::EMCAL_COLS - 1)
+      towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta + 1);
+    if (ieta > 0)
+      towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta - 1);
+  }
+
+  LOG(debug) << "iSM " << iSM << ", towerId " << towerId << ", a " << towerId1 << ", b " << towerId2 << ", c " << towerId3 << ", e " << towerId3;
+
+  float ecell1 = 0, ecell2 = 0, ecell3 = 0, ecell4 = 0;
+  float tcell1 = 0, tcell2 = 0, tcell3 = 0, tcell4 = 0;
+
+  short iTowerId = -1;
+  for (auto const& cell : mInputsContainer) {
+    iTowerId = cell.getTower();
+    if (towerId1 == iTowerId) {
+      ecell1 = cell.getEnergy();
+      tcell1 = cell.getTimeStamp();
+    }
+    if (towerId2 == iTowerId) {
+      ecell2 = cell.getEnergy();
+      tcell2 = cell.getTimeStamp();
+    }
+    if (towerId3 == iTowerId) {
+      ecell3 = cell.getEnergy();
+      tcell3 = cell.getTimeStamp();
+    }
+    if (towerId4 == iTowerId) {
+      ecell4 = cell.getEnergy();
+      tcell4 = cell.getTimeStamp();
+    }
+  }
+
+  if (std::abs(exoticTime - tcell1) > mExoticCellDiffTime)
+    ecell1 = 0;
+  if (std::abs(exoticTime - tcell2) > mExoticCellDiffTime)
+    ecell2 = 0;
+  if (std::abs(exoticTime - tcell3) > mExoticCellDiffTime)
+    ecell3 = 0;
+  if (std::abs(exoticTime - tcell4) > mExoticCellDiffTime)
+    ecell4 = 0;
+
+  float w1 = 1, w2 = 1, w3 = 1, w4 = 1;
+  if (mUseWeightExotic) {
+    w1 = GetCellWeight(ecell1, energy);
+    w2 = GetCellWeight(ecell2, energy);
+    w3 = GetCellWeight(ecell3, energy);
+    w4 = GetCellWeight(ecell4, energy);
+  }
+
+  if (ecell1 < mExoticCellInCrossMinAmplitude || w1 <= 0)
+    ecell1 = 0;
+  if (ecell2 < mExoticCellInCrossMinAmplitude || w2 <= 0)
+    ecell2 = 0;
+  if (ecell3 < mExoticCellInCrossMinAmplitude || w3 <= 0)
+    ecell3 = 0;
+  if (ecell4 < mExoticCellInCrossMinAmplitude || w4 <= 0)
+    ecell4 = 0;
+
+  return ecell1 + ecell2 + ecell3 + ecell4;
+}
+
+///
+/// return weight of cell for shower shape calculation
+//____________________________________________________________________________
+template <class InputType>
+float ClusterFactory<InputType>::GetCellWeight(float eCell, float eCluster) const
+{
+  if (eCell > 0 && eCluster > 0) {
+    if (mLogWeight > 0)
+      return std::max(0.f, mLogWeight + std::log(eCell / eCluster));
+    else
+      return std::log(eCluster / eCell);
+  } else
+    return 0.;
 }
 
 ///

--- a/Detectors/EMCAL/base/src/ClusterFactory.cxx
+++ b/Detectors/EMCAL/base/src/ClusterFactory.cxx
@@ -618,10 +618,12 @@ float ClusterFactory<InputType>::getECross(short towerId, float energy, float co
   short towerId1 = -1;
   short towerId2 = -1;
 
-  if (iphi < o2::emcal::EMCAL_ROWS - 1)
+  if (iphi < o2::emcal::EMCAL_ROWS - 1) {
     towerId1 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi + 1, ieta);
-  if (iphi > 0)
+  }
+  if (iphi > 0) {
     towerId2 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi - 1, ieta);
+  }
 
   // In case of cell in eta = 0 border, depending on SM shift the cross cell index
 
@@ -635,65 +637,75 @@ float ClusterFactory<InputType>::getECross(short towerId, float energy, float co
     towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta + 1);
     towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM - 1, iphi, o2::emcal::EMCAL_COLS - 1);
   } else {
-    if (ieta < o2::emcal::EMCAL_COLS - 1)
+    if (ieta < o2::emcal::EMCAL_COLS - 1) {
       towerId3 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta + 1);
-    if (ieta > 0)
+    }
+    if (ieta > 0) {
       towerId4 = mGeomPtr->GetAbsCellIdFromCellIndexes(iSM, iphi, ieta - 1);
+    }
   }
 
   LOG(debug) << "iSM " << iSM << ", towerId " << towerId << ", a " << towerId1 << ", b " << towerId2 << ", c " << towerId3 << ", e " << towerId3;
 
-  float ecell1 = 0, ecell2 = 0, ecell3 = 0, ecell4 = 0;
-  float tcell1 = 0, tcell2 = 0, tcell3 = 0, tcell4 = 0;
+  std::vector<float> ecell(4, 0.);
+  std::vector<float> tcell(4, 0.);
 
   short iTowerId = -1;
   for (auto const& cell : mInputsContainer) {
     iTowerId = cell.getTower();
     if (towerId1 == iTowerId) {
-      ecell1 = cell.getEnergy();
-      tcell1 = cell.getTimeStamp();
+      ecell[0] = cell.getEnergy();
+      tcell[0] = cell.getTimeStamp();
     }
     if (towerId2 == iTowerId) {
-      ecell2 = cell.getEnergy();
-      tcell2 = cell.getTimeStamp();
+      ecell[1] = cell.getEnergy();
+      tcell[1] = cell.getTimeStamp();
     }
     if (towerId3 == iTowerId) {
-      ecell3 = cell.getEnergy();
-      tcell3 = cell.getTimeStamp();
+      ecell[2] = cell.getEnergy();
+      tcell[2] = cell.getTimeStamp();
     }
     if (towerId4 == iTowerId) {
-      ecell4 = cell.getEnergy();
-      tcell4 = cell.getTimeStamp();
+      ecell[3] = cell.getEnergy();
+      tcell[3] = cell.getTimeStamp();
     }
   }
 
-  if (std::abs(exoticTime - tcell1) > mExoticCellDiffTime)
-    ecell1 = 0;
-  if (std::abs(exoticTime - tcell2) > mExoticCellDiffTime)
-    ecell2 = 0;
-  if (std::abs(exoticTime - tcell3) > mExoticCellDiffTime)
-    ecell3 = 0;
-  if (std::abs(exoticTime - tcell4) > mExoticCellDiffTime)
-    ecell4 = 0;
+  if (std::abs(exoticTime - tcell[0]) > mExoticCellDiffTime) {
+    ecell[0] = 0;
+  }
+  if (std::abs(exoticTime - tcell[1]) > mExoticCellDiffTime) {
+    ecell[1] = 0;
+  }
+  if (std::abs(exoticTime - tcell[2]) > mExoticCellDiffTime) {
+    ecell[2] = 0;
+  }
+  if (std::abs(exoticTime - tcell[3]) > mExoticCellDiffTime) {
+    ecell[3] = 0;
+  }
 
   float w1 = 1, w2 = 1, w3 = 1, w4 = 1;
   if (mUseWeightExotic) {
-    w1 = GetCellWeight(ecell1, energy);
-    w2 = GetCellWeight(ecell2, energy);
-    w3 = GetCellWeight(ecell3, energy);
-    w4 = GetCellWeight(ecell4, energy);
+    w1 = GetCellWeight(ecell[0], energy);
+    w2 = GetCellWeight(ecell[1], energy);
+    w3 = GetCellWeight(ecell[2], energy);
+    w4 = GetCellWeight(ecell[3], energy);
   }
 
-  if (ecell1 < mExoticCellInCrossMinAmplitude || w1 <= 0)
-    ecell1 = 0;
-  if (ecell2 < mExoticCellInCrossMinAmplitude || w2 <= 0)
-    ecell2 = 0;
-  if (ecell3 < mExoticCellInCrossMinAmplitude || w3 <= 0)
-    ecell3 = 0;
-  if (ecell4 < mExoticCellInCrossMinAmplitude || w4 <= 0)
-    ecell4 = 0;
+  if (ecell[0] < mExoticCellInCrossMinAmplitude || w1 <= 0) {
+    ecell[0] = 0;
+  }
+  if (ecell[1] < mExoticCellInCrossMinAmplitude || w2 <= 0) {
+    ecell[1] = 0;
+  }
+  if (ecell[2] < mExoticCellInCrossMinAmplitude || w3 <= 0) {
+    ecell[2] = 0;
+  }
+  if (ecell[3] < mExoticCellInCrossMinAmplitude || w4 <= 0) {
+    ecell[3] = 0;
+  }
 
-  return ecell1 + ecell2 + ecell3 + ecell4;
+  return ecell[0] + ecell[1] + ecell[2] + ecell[3];
 }
 
 ///
@@ -703,12 +715,14 @@ template <class InputType>
 float ClusterFactory<InputType>::GetCellWeight(float eCell, float eCluster) const
 {
   if (eCell > 0 && eCluster > 0) {
-    if (mLogWeight > 0)
+    if (mLogWeight > 0) {
       return std::max(0.f, mLogWeight + std::log(eCell / eCluster));
-    else
+    } else {
       return std::log(eCluster / eCell);
-  } else
+    }
+  } else {
     return 0.;
+  }
 }
 
 ///

